lib/src/rules/directives_ordering.dart

```diff
@@ -373,3 +373,17 @@ class _Visitor extends SimpleAstVisitor<void> {
   Iterable<ImportDirective> _getImportDirectives(CompilationUnit node) =>
       node.directives.whereType<ImportDirective>();
 }
+
+/// Compares directives by package then file in package.
+///
+/// Package is everything until the first `/`.
+int compareDirectives(String a, String b) {
+  if (!a.contains('/') || !b.contains('/')) {
+    return a.compareTo(b);
+  }
+  var as = a.split('/');
```

Comment:
Instead of using `String.split`, consider something like
```
var indexA = a.indexOf('/');
var indexB = B.indexOf('/');
var result = a.substring(indexA).compareTo(b.substring(indexB));
if (result != 0) return result;
return a.substring(indexA + 1).compareTo(b.substring(indexB + 1));
```
That will involve creating fewer temporary strings and avoid the need to join them all back together at the end.

I don't know whether there would be a significant performance improvement, but it seems like a reasonable optimization.

Reply:
Yes, please.  I think this is great idea.

> I don't know whether there would be a significant performance improvement, but it seems like a reasonable optimization.

Every little bit counts! 👍

Reply:
Yeah, I was fairly surprised by how much of a performance improvement similar changes were able to get us in code completion.

Reply:
Done.